rs/consensus/utils/src/lib.rs

```diff
@@ -272,9 +272,17 @@ pub fn get_adjusted_notary_delay_from_settings(
     NotaryDelay::CanNotarizeAfter(Duration::from_millis(certified_adjusted_delay))
 }
 
-/// Return the validated block proposals with the lowest rank at height `h`, if
-/// there are any. Else return `None`.
+/// Return the validated block proposals with the lowest rank at height `h` that
+/// has not been disqualified, if there are any. Else return `None`.
 pub fn find_lowest_ranked_proposals(pool: &PoolReader<'_>, h: Height) -> Vec<BlockProposal> {
```

Comment:
Maybe we should consider renaming this function to something like `find_lowest_ranked_non_disqualified_proposals`

Reply:
done